File: src/boj/boj_10845.java
package boj;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class boj_10845 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        Deque<Integer> deque = new ArrayDeque<>();
        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String ex = st.nextToken();

            switch (ex) {
                case "push":
                    deque.add(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    bw.append(deque.isEmpty() ? "-1" : String.valueOf(deque.poll())).append('\n');
                    break;
                case "size":
                    bw.append(String.valueOf(deque.size())).append('\n');
                    break;
                case "empty":
                    bw.append(deque.isEmpty() ? '1' : '0').append('\n');
                    break;
                case "front":
                    bw.append(deque.isEmpty() ? "-1" : String.valueOf(deque.peekFirst())).append('\n');
                    break;
                case "back":
                    bw.append(deque.isEmpty() ? "-1" : String.valueOf(deque.peekLast())).append('\n');
                    break;
            }
        }
        bw.close();
        br.close();
    }
}

File: src/boj/boj_10845.kt
package boj_10845

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val queue = Queue()
    repeat(readLine().toInt()) {
        with(StringTokenizer(readLine())) {
            when (nextToken()) {
                "push" -> queue.push(nextToken().toInt())
                "pop" -> bufferedWriter.append("${queue.pop()}\n")
                "size" -> bufferedWriter.append("${queue.size()}\n")
                "empty" -> bufferedWriter.append("${queue.empty()}\n")
                "front" -> bufferedWriter.append("${queue.front()}\n")
                "back" -> bufferedWriter.append("${queue.back()}\n")
                else -> throw Exception()
            }
        }
    }
    bufferedWriter.flush()
    bufferedWriter.close()
    close()
}

class Queue {
    class Node(val value: Int) {
        var next: Node? = null
    }

    private var root: Node? = null
    private var size = 0

    fun push(n: Int) {
        if (size == 0) {
            root = Node(n)
        } else {
            var node = root
            while (node!!.next != null) {
                node = node.next
            }
            node.next = Node(n)
        }
        size++
    }

    fun pop(): Int {
        if (size == 0) {
            return -1
        }
        size--
        val result = root!!.value
        root = root!!.next
        return result
    }

    fun size(): Int {
        return size
    }

    fun empty(): Int {
        return if (size == 0) {
            1
        } else {
            0
        }
    }

    fun front(): Int {
        if (size == 0) {
            return -1
        }
        return root!!.value
    }

    fun back(): Int {
        if (size == 0) {
            return -1
        }
        var node = root!!
        while (node.next != null) {
            node = node.next!!
        }
        return node.value
    }
}
